Cargo.toml

```diff
@@ -42,13 +42,15 @@ tracing-metrics = { path = "lib/tracing-metrics" }
 tracing-limit = { path = "lib/tracing-limit" }
 
 # Tokio / Futures
-futures = "0.1.25"
-futures03 = { package = "futures", version = "0.3", default-features = false, features = ["compat"] }
+futures01 = { package = "futures", version = "0.1.25" }
```

Comment:
I'd like to keep this as `futures` for now and change `futures03` back to `futures` once we are fully upgraded. 

if we want to rename futures01 we should do that in a separate PR.

Reply:
I do think it'd be worth to rename to `futures01` before we're totally done. Mostly because people will be less likely to use it accidentally, and also so we can easily grep for it to find places that still need upgraded.

Reply:
I am happy to change that but I think these migration PRs should remain small and focused. So for me I'd rather see a follow up PR change the name of futures to futures01.

Reply:
I was thinking this change marks an "official" start of the migration to futures 0.3. Doing it at the start should let us eventually get to a situation where we don't have any `futures01` imports left. If we do it in the reverse order, we'll have to do a similar massive diff after the transition - to rename `futures03` to `futures`.
I'd prefer this sooner than later. After all, it's a pending thing that we'll have to do eventually, why leave it for tomorrow if we can do it today.

Reply:
> I am happy to change that but I think these migration PRs should remain small and focused. So for me I'd rather see a follow up PR change the name of futures to futures01.

I can definitely split this into a separate PR!

Reply:
Well mostly, the crate name change makes this diff noisy and I want to ensure that runtime change is not a noisy change but a very focused one. I don't think it hurts to open a follow up PR with that change :man_shrugging:

In the past through my upgrade experience the noisier the change the more likely we are to introduce slight bugs. like this one https://github.com/tower-rs/tower/issues/415 which I introduced and was not caught since the diffs were large.

We can do another PR or if its easier split the commits in this PR so I can review the runtime changes then the module name change.

Reply:
I created the following PRs:
- #1933
- #1934 

I'll also create some more.

Reply:
> Well mostly, the crate name change makes this diff noisy and I want to ensure that runtime change is not a noisy change but a very focused one.

This is fine with me, but I do think it's worth differentiating between noisy and large PRs. Changing the name everywhere makes for a larger diff, but as a mechanical name change, it can still be quick and easy to review. Isolating mechanical changes to a single commit within a PR is a good option too, since reviewers know it's safe to skim.

Reply:
In this case, it made a lot of sense to extract the change, cause we had the ability to split the concerns. We were able to merge that one in much quicker than if it'd be a part of this PR, and I think it's a good result. So I think we found a good balance here.
At the same time, isolating mechanical changes as a commit works too! This change ended up being particularly tricky, so we naturally came to splitting the original PR with rename isolated as commit - so I guess we used both approaches here. :smile: